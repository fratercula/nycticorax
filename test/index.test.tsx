import React from 'react'
import { mount } from 'enzyme'
import Nycticorax from '../src'

type Store = { name: string, age: number }

const nycticorax = new Nycticorax<Store>()

const {
  createStore,
  // dispatch,
  // connect,
  useStore,
} = nycticorax

// type Connect = CT<Store>

createStore({ name: 'xyz', age: 1 })

describe('nycticorax', () => {
  it('hook', () => {
    function Hook() {
      const { name, age } = useStore('name', 'age')
      return (
        <div>
          <div>{name}</div>
          <div>{age}</div>
        </div>
      )
    }
    const wrapper = mount(<Hook />)
    expect(wrapper.render()).toMatchSnapshot()
  })
})
