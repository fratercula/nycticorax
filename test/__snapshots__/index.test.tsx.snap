// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nycticorax hook 1`] = `
<div>
  <div>
    xyz
  </div>
  <div>
    1
  </div>
</div>
`;
